src/pymor/analyticalproblems/expressions.py

```diff
@@ -58,6 +58,12 @@
 builtin_max = max
 
 
+if np.lib.NumpyVersion(np.__version__) >= '2.0.0':
+    NUMPY_COPY_IF_NEEDED = None
+elif np.lib.NumpyVersion(np.__version__) < '1.28.0':
+    NUMPY_COPY_IF_NEEDED = False
+
```

Comment:
Would you not want an `else` branch here? Or do you want the call below to fail on purpose for unsuitable numpy versions (>= 1.28 && < 2) to detect a mismatch in the hardcoded versions here w.r.t. the ones in the pyproject.toml?

Reply:
Guess I messed up when adapting this from scipy. Fixed.